frontend/src/pages/user/ModalAddUser.js

```diff
@@ -0,0 +1,311 @@
+import {useEffect, useState} from 'react';
+import Modal from "react-bootstrap/Modal";
+import Form from "react-bootstrap/Form";
+import Button from "react-bootstrap/Button";
+import ErrorMessage from "../../messages/errorMessage";
+
+function ModalAddUser(props) {
+
+
+    const currentCustomerId = localStorage.getItem("currentCustomerId") != null ? localStorage.getItem("currentCustomerId") : 0;
+
+    const [state, setState] = useState([]);
+
+    const [states, setStates] = useState([]);
+
+    const [addressDto, setAddressDto] = useState({
+        city: '',
+        state: null,
+        addressLine1: '',
+        addressLine2: ''
+    });
+    const [userDto, setUser] = useState({
+        name: '',
+        surname: '',
+        birthday: '',
+        addressDto: addressDto,
+        role: 'ROLE_SYSTEM_ADMIN',
+        username: '',
+        email: '',
+        password:'',
+        customerId: currentCustomerId
+    });
+    const [validError, setError] = useState([]);
+    const [errorMessage, setErrors] = useState('');
+
+    const onChangeState = (e) => {
+        const selectedState = states.find(state => state.state === e.target.value);
+
+        setState(preState => ({
+            ...preState,
+            state: selectedState
+        }));
+        console.log(selectedState);
+        setAddressDto(preState => ({
+            ...preState,
+            state: selectedState
+        }));
+    };
+    useEffect(() => {
+
+
+        fetch('/states')
+            .then(response => response.json())
+            .then(commits => {
+                setStates(commits.content);
+            });
+    }, []);
+
+    const handleName = (e) => {
+        setUser(preState => ({
+            ...preState,
+            name: e.target.value
+        }));
+    };
+    const handlePassword = (e) => {
+        setUser(preState => ({
+            ...preState,
+            password: e.target.value
+        }));
+    };
+    const handleSurname = (e) => {
+        setUser(preState => ({
+            ...preState,
+            surname: e.target.value
+        }));
+    };
+    const handleUsername = (e) => {
+        setUser(preState => ({
+            ...preState,
+            username: e.target.value
+        }));
+    };
+    const handleBirthday = (e) => {
+        setUser(preState => ({
+            ...preState,
+            birthday: e.target.value
+        }));
+    };
+    const handleRole = (e) => {
+        console.log(e.target.value);
+        setUser(preState => ({
+            ...preState,
+            role: e.target.value
+        }));
+    };
+    const handleEmail = (e) => {
+        setUser(preState => ({
+            ...preState,
+            email: e.target.value
+        }));
+    };
+    const handleCity = (e) => {
+        setAddressDto(preState => ({
+            ...preState,
+            city: e.target.value
+        }));
+    };
+    const handleaddressLine1 = (e) => {
```

Comment:
This is the general observation. HandleSomething is very general name. Here we catch the change of city. So I would name it like "onCityChange". It may be "onCitySelect", or in other case like "onCityClick". And this is applicable to others as well. It may be "handleCItyChange", why not?